Sources/StreamChat/StateLayer/UserSearchState.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Represents a list of user search results.
+@available(iOS 13.0, *)
+public final class UserSearchState: ObservableObject {
+    /// The search query the results corresponds to.
+    @Published public private(set) var searchQuery: UserListQuery?
+    
+    /// An array of search results for the specified query and pagination state.
+    @Published public private(set) var users = StreamCollection<ChatUser>([])
+    
+    // MARK: - Mutating the State
+    
+    @MainActor func setSearchQuery(_ query: UserListQuery) {
+        searchQuery = query
+    }
+    
+    @MainActor func setUsers(_ newUsers: [ChatUser], for pagination: Pagination) {
+        if pagination.offset > 0 {
+            var result = Array(users[..<pagination.offset])
+            result.append(contentsOf: newUsers)
+            users = StreamCollection(result)
+        } else {
+            users = StreamCollection(newUsers)
+        }
+    }
```

Comment:
hmm, this is strange - can you please explain why we need this? I don't think we had such logic before.

Reply:
`ChatUserSearchController` uses CoreData for driving the results array like all the other controllers. Since in this PR I went for skipping the CoreData part. The downside in that case is that then I need to take care of managing the users array myself since any of the fetched results are never put into the DB (which makes sense for searching).
I'll think a little bit more about this, sure I can make it nicer and more reliable.